### PR TITLE
feat(FX-4684): Event schema for Artwork Lists user interactions

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feat**

Co-authored-by: Abhiti Prabahar <abhiti.prabahar@gmail.com>

This PR resolves [FX-4684]

Events: [Google Spreadsheet](https://docs.google.com/spreadsheets/d/1_VDAmRJz2K87FXDq7oh_tQSgwBv4syuZjaaR3sfetcI/edit#gid=1013464765)

### Added events
* `addedArtworkToArtworkList`
* `createdArtworkList`
* `viewedArtworkList`
* `deletedArtworkList`
* `editedArtworkList`

### PR Checklist (tick all before merging)

- [x] If I've added a new file to the tree I've exported it from the common [`index.ts`](https://github.com/artsy/cohesion/blob/main/src/Schema/Events/index.ts)
- [x] I've added comments with examples for any new interfaces and ensured that they're in the docs
- [x] No platform-specific terminology has been used outside of `click` and `tap` (platform is inferred by the DB storing events)

[FX-4684]: https://artsyproduct.atlassian.net/browse/FX-4684?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Schema/Events/Lists.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+/**
+ * When a user adds an artwork to an artwork list
+ *
+ * This schema describes events sent to Segment from [[addedArtworkToArtworkList]]
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "addedArtworkToArtworkList",
```
Comment (dimatretyak) [inline src/Schema/Events/Lists.ts:17]: **question**:  do we need to use `ArtworkList` in events or just `List` (e.g. `addedArtworkToList`). Based on [this Notion Card](https://www.notion.so/artsy/Artworks-list-I-think-we-ve-got-a-confusing-mix-of-terms-in-our-codebase-for-Artwork-Lists-collec-03de616986ab4cd4a839ec6aed989d3d)

cc: @anandaroop @nickskalkin

File: src/Schema/Events/Lists.ts
[the remark this answers is not in the document]
Comment (abhitip) [reply src/Schema/Events/Lists.ts:1]: I think `ArtworkList` would make more sense to me!

File: src/Schema/Events/Lists.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ *    context_owner_id: "641b795ae11bda000c80d58d",
+ *    context_owner_slug: "banksy-gangsta-rat-unsigned-49",
+ *    context_owner_type: "artwork",
+ *    artwork_list_name: "List #1"
+ *  }
+ * ```
+ */
+export interface AddedArtworkToArtworkList {
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/Schema/Events/Lists.ts:25]: **question**: Should we track this event when a user selects artworks after creating a list (see [this flow in Figma](https://www.figma.com/file/wSV2KNz97Tyj8HkH2ZYAkB/Save-%2B-Lists?node-id=308-219486&t=DqwhyvA1n5RgGc58-0))? If so, do we have to track this event for each artwork? For example, if 5 artworks are selected, then should we track the `AddedArtworkToArtworkList` event 5 times?

cc: @nickskalkin @anandaroop

[the remark this answers is not in the document]
Comment (anandaroop) [reply src/Schema/Events/Lists.ts:25]: Yeah in general, depending on the context we are either:

- adding 1 artwork to N lists (e.g. from an artwork page)
- adding N artworks to 1 list (e.g. from a newly created list)

Not sure how much of this needs to be conveyed in the event vs analysis in redshift — I'll let @abhitip weigh in!

File: src/Schema/Events/Lists.ts (hunk 1 above)
Comment (dimatretyak) [reply src/Schema/Events/Lists.ts:17]: [Discussed and decided](https://artsy.slack.com/archives/C9SATFLUU/p1680012458469089?thread_ts=1680012130.077609&cid=C9SATFLUU) to use` ArtworkList` term

File: src/Schema/Events/Lists.ts
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/Schema/Events/Lists.ts:1]: Mark as resolved (see [this message](https://artsy.slack.com/archives/C9SATFLUU/p1680012458469089?thread_ts=1680012130.077609&cid=C9SATFLUU)). Thank y'all 🙏

File: src/Schema/Events/Lists.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
+import { OwnerType } from "../Values/OwnerType"
+import { ActionType } from "."
+
+/**
+ * Schema describing 'Lists' events
+ * @packageDocumentation
+ */
+
+/**
+ * When a user adds an artwork to an artwork list
```
Comment (dimatretyak) [inline src/Schema/Events/Lists.ts:10]: question: @abhitip the user can also delete artwork from the list. do we have to track the event for this?

File: src/Schema/Events/Lists.ts (hunk 2 above)
Comment (dimatretyak) [reply src/Schema/Events/Lists.ts:25]: **suggestion**

#### adding 1 artwork to N lists (e.g. from an artwork page)
```javascript
{
  action: "addedArtworkToArtworkList",
  context_owner_id: "641b795ae11bda000c80d58d",
  context_owner_slug: "banksy-gangsta-rat-unsigned-49",
  context_owner_type: "artwork",
  artwork_ids: ["artwork-id-one"],
  owner_ids: ["artwork-list-id-one", "artwork-list-id-two"]
}
```

#### adding N artworks to 1 list (e.g. from a newly created list)
```javascript
{
  action: "addedArtworkToArtworkList",
  context_owner_id: "641b795ae11bda000c80d58d",
  context_owner_slug: "banksy-gangsta-rat-unsigned-49",
  context_owner_type: "artwork",
  artwork_ids: ["artwork-id-one", "artwork-id-two"],
  owner_ids: ["artwork-list-id-one"]
}
```

Comment (dimatretyak) [reply src/Schema/Events/Lists.ts:26]: Done: https://github.com/artsy/cohesion/pull/421/commits/f8ad3559c75d8eefddcdf08f337b857a932e2ccc

Comment (dimatretyak) [reply src/Schema/Events/Lists.ts:26]: Done: https://github.com/artsy/cohesion/pull/421/commits/f8ad3559c75d8eefddcdf08f337b857a932e2ccc

**Comment by artsyit** [comment]
:rocket: PR was released in [`v4.120.0`](https://github.com/artsy/cohesion/releases/tag/v4.120.0) :rocket:

**Pull request merged**